src/measure_theory/arithmetic.lean

```diff
@@ -285,19 +287,29 @@ measurable_inv.comp_ae_measurable hf
 
 end inv
 
-instance has_measurable_gpow (G : Type*) [group G] [measurable_space G]
+/- There is something extremely strange here: copy-pasting the proof of this lemma in the proof
+of `has_measurable_gpow` fails, while `pp.all` does not show any difference in the goal.
+Keep it as a separate lemmas as a workaround. -/
```

Comment:
I had a play with this too, and am also stumped.

Reply:
I am planning to mention this on Zulip once the PR is merged, so that everyone can play with it and investigate. It feels like a genuine Lean bug, the first I've seen in a long time.